batch/batch/worker/worker.py

```diff
@@ -474,17 +478,28 @@ async def localize_rootfs():
             self.state = 'error'
             self.error = traceback.format_exc()
         finally:
-            with self.step('deleting'):
-                await self.delete_container()
+            await self.delete_container()
+            if self.image_digest:
+                worker.image_digest_ref_count[self.image_digest] -= 1
+                assert worker.image_digest_ref_count[self.image_digest] >= 0
```

Comment:
You might want to add what to print if the assert fails. Not sure if it's helpful or not.

```
assert worker.image_digest_ref_count[self.image_digest] >= 0, f'{worker.image_digest_ref_count}'
```